Sources/Client/Client/Client.swift

```diff
@@ -18,16 +18,62 @@ public final class Client: Uploader {
     public typealias OnEvent = (Event) -> Void
     
     /// A client config (see `Config`).
-    public static var config = Config(apiKey: "")
-    /// A shared client.
-    public static let shared = Client()
+    @available(*, deprecated, message: """
+    Configuring the shared Client using the static `Client.config` variable has been deprecated. Please create an instance
+    of the `Client.Config` struct and call `Client.configure(_:)` to set up the shared instance of Client.
+    """)
+    public static var config: Config = .init(apiKey: "_deprecated") {
+        didSet {
+            guard _sharedClient == nil else {
+                assertionFailure("`Client.shared` instance was already used. It's not possible to change its configuration.")
+                return
+            }
+
+            configForSharedClient = config
+        }
+    }
+
+    /// The configuration object used for creating `Client.shared`.
+    private static var configForSharedClient: Config?
+
+    /// The shared client.
+    public static var shared: Client {
+        if let client = _sharedClient {
+            // Return the existing instance
+            return client
+        }
+
+        assert(
+            configForSharedClient != nil,
+            "The shared instance of the Stream chat client wasn't configured. " +
+            "Create an instance of the `Client.Config` struct and call `Client.configure(_:)` to set it up."
+        )
+
+        _sharedClient = Client(configuration: configForSharedClient ?? .init(apiKey: "_deprecated"))
```

Comment:
Why aren't we using empty API key here? So it'll trigger another assert (if possible)

Reply:
I don't know to be honest. It felt wrong to write code that I know will cause assertion failure.

Reply:
Only if our SDK is not used correctly though, right?

Reply:
If you don't like `"_deprecated"` we can use something else (please suggest) but using `""` feels wrong, sorry 🤷‍♂️

Reply:
It's not like I don't like `_deprecated`, but these asserts will not run in release builds and there can be a client trying to connect with `_deprecated` api key. This code will only run when shared client is touched, and it's an error to touch it before configuring it correctly, so I think `""` is good here - user didn't configure it, so it's empty.

Reply:
So what about: `" __ API KEY NOT CONFIGURED __"`?

Reply:
mb `__API_KEY_NOT_CONFIGURED__`? :)